### PR TITLE
Montana Personal Income Tax Schedule

### LINKED ISSUE #2264: Montana Personal Income Tax Schedule
[Reference](https://leg.mt.gov/bills/mca/title_0150/chapter_0300/part_0210/section_0030/0150-0300-0210-0030.html)

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #2264

### CODE REVIEW

File: policyengine_us/variables/gov/states/mt/tax/income/mt_income_tax_before_refundable_credits.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    label = "Montanta income tax before refunable credits"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        income = tax_unit("mt_taxable_income", period)
+        p = parameters(period).gov.states.mt.tax.income.main
+        # Before 2024 Montana has a flat tax rate for each filing status
+        temporary_rate = p.temporary_rate.calc(income)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_income_tax_before_refundable_credits.py:16]: More verbose but I'd suggest putting the non-filing-status-specific rates in the filing status files to make the Python simpler.

File: policyengine_us/parameters/gov/states/mt/tax/income/main/progressive/head_of_household.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+description: Montana provides the following marginal tax rate for the head of the household.
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: Montana income tax rate head of household
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/main/progressive/head_of_household.yaml:6]:
```suggestion
  label: Montana income tax rates for head of household filers
```

File: policyengine_us/parameters/gov/states/mt/tax/income/main/progressive/head_of_household.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+description: Montana provides the following marginal tax rate for the head of the household.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/main/progressive/head_of_household.yaml:1]: And elsewhere
```suggestion
description: Montana taxes head of household filers at this rate structure.
```

File: policyengine_us/variables/gov/states/mt/tax/income/mt_income_tax_before_refundable_credits.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+from policyengine_us.model_api import *
+
+
+class mt_income_tax_before_refundable_credits(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Montanta income tax before refunable credits"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_income_tax_before_refundable_credits.py:7]: Montana typo elsewhere
```suggestion
    label = "Montana income tax before refundable credits"
```

File: policyengine_us/parameters/gov/states/mt/tax/income/main/head_of_household.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
+description: Montana taxes head of household filers at this rate structure.
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: Montana income tax rates for head of household filers
+  reference:
+    - title: Montana Code Annotated 2021 Title 15, Chapter 30, Part 21, 15-30-2103 
+      href: https://leg.mt.gov/bills/mca/title_0150/chapter_0300/part_0210/section_0030/0150-0300-0210-0030.html
+# Before 2024 Montana has a flat tax rate for each filing status
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/main/head_of_household.yaml:10]: Not a flat rate right, just the same across filing statuses?

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mt/tax/income/main/head_of_household.yaml]: correct, I changed the formatting

**Pull request merged**